tool/functiontool/function.go

```diff
@@ -140,7 +140,10 @@ func (f *functionTool[TArgs, TResults]) Run(ctx tool.Context, args any) (map[str
 	if err != nil {
 		return nil, err
 	}
-	output := f.handler(ctx, input)
+	output, err := f.handler(ctx, input)
+	if err != nil {
+		return nil, err
```

Comment:
Now we can remove the TODO :-) in https://github.com/google/adk-go/blob/cffe3b0d6b8b45505a67f658d9cf941849d14b10/internal/llminternal/base_flow.go#L422